### PR TITLE
RATEST-276: Move heroku deploy config to the parent root

### PR DESCRIPTION
**Pull request by kdaud**
Ticket [ID](https://issues.openmrs.org/browse/RATEST-276)

Description -> Move heroku deploy config to the parent root

### CODE REVIEW

**Comment by kdaud** [comment]
@dkayiwa @sherrif10 we would need to move the heroku configs from a submodule to the parent root. 

However, am failing to figure out how to ensure a [repapp webui ](https://hub.docker.com/r/openmrs/openmrs-reference-application-distro/) is deployed within the Heroku pipeline so that PR changes are tested out on webui before they are merged.
Some ideas around the subject?

**Comment by kdaud** [comment]
Have gone through a particular heroku [documentation](https://devcenter.heroku.com/articles/build-docker-images-heroku-yml) and the idea seems to be around building a docker image for refapp webui within heroku.yml file. Not sure whether this idea could be a better in regard to what the ticket is addressing.

**Comment by dkayiwa** [comment]
> the idea seems to be around building a docker image for refapp webui within heroku.yml file. Not sure whether this idea could be a better approach !

Yes that is the way to go.

**Comment by kdaud** [comment]
> Yes that is the way to go.

Thanks @dkayiwa for the hint which have been thinking of would be the way forward.

**Comment by sherrif10** [comment]
Great, @kdaud  . so i think you add some configs in heroku.yml for the current docker image to spin up the instance.

**Comment by kdaud** [comment]
@sherrif10 have added the config file and might be missing some configs to have the docker file for refapp webui successfully deployed. do you mind looking through the file and let me know what I may need to add?

File: heroku.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+setup:
+ addons:
+   - plan: heroku-postgresql
+build:
+ docker:
+   web: docker pull openmrs/openmrs-reference-application-distro:latest
```
Comment (sherrif10) [inline heroku.yml:6]: what happens when you docker compose up .in this directory

Comment (kdaud) [reply heroku.yml:6]: do you mean to just have the docker file within the repo  and point to its location instead?

**Comment by kdaud** [comment]
@dkayiwa any review comment?

**Comment by dkayiwa** [comment]
@kdaud does it work?

**Comment by kdaud** [comment]
Have just noticed I have to add a docker file for refapp webui and instead point the web config to its directory.

**Comment by sherrif10** [comment]
@kdaud  let me know where i can provide support

**Comment by kdaud** [comment]
@dkayiwa @sherrif10 this work is ready for review !

**Comment by sherrif10** [comment]
is this going to be automatically tiggered on each single push or its going to be manually triggered ?

**Comment by kdaud** [comment]
auto

**Comment by dkayiwa** [comment]
@kdaud have you confirmed that it works as advertised?

**Comment by kdaud** [comment]
> @kdaud have you confirmed that it works as advertised?

I think so!

**Comment by kdaud** [comment]
Testing out the pipeline requires when these changes are in master so that I tested it out on an active PR.

**Comment by dkayiwa** [comment]
> I think so!

This meaneth that you are sureth not!

**Comment by kdaud** [comment]
@dkayiwa testing out the pipeline requires when these changes are in master so that I test it out on an active PR.

**Comment by dkayiwa** [comment]
ok can you now test?

**Comment by kdaud** [comment]
Have tested out the app with pull/319 and got out this [result](https://pastebin.com/NaPwZwcH). The build result lookis similar to what Bamboo would show after merging the PR however, it seems like the refapp webui is not being deployed on heroku  regardless of adding [heroku.yml](https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/heroku.yml) config file. Any idea on the subject?

**Comment by dkayiwa** [comment]
Which heroku documentation are you using to set this up?

**Comment by kdaud** [comment]
Pertaining heroki set up, I

1. set up the [static.json](https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/static.json) file that cdefines details on how the Heroku PaaS should serve each static build and the [app.json](https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/app.json) file that defines details of how each app should be setup.
2. then added [heroku ](https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/heroku.yml)file to build the docker Image for refapp web ui.


@dkayiwa could there be something am missing out that I need to do in order to have the tool functional?

**Comment by dkayiwa** [comment]
@kdaud what happened to what you demonstrated to us where you would deploy changes in a pull request and the only problem being that the deployment was not automatic? It had to be manually triggered by a click of a button.

**Comment by kdaud** [comment]
That pipeline still exists and works well. One thing though is that the heroku account is personal so not every one can access the pipeline. The build result that is given when the changes are deployed on the heroku is the same with what Bamboo would produce when the PR changes are merged into master.

**Pull request merged**